tests/topotests/bgp_evpn_vxlan_macvrf_soo_topo1/test_bgp_evpn_vxlan_macvrf_soo.py

```diff
@@ -0,0 +1,839 @@
+    pe1.run("ip link set dev br101 addr aa:bb:cc:00:11:aa")
+    pe1.run("ip link set dev br101 master VRF-A")
+    pe1.run("ip link set dev br101 up")
+    pe1.run(
+        "ip link add vxlan101 type vxlan id 101 dstport 4789 local 10.10.10.10 nolearning"
+    )
+    pe1.run("ip link set dev vxlan101 master br101")
+    pe1.run("ip link set dev vxlan101 type bridge_slave neigh_suppress on learning off")
+    pe1.run("ip link set dev vxlan101 up")
+    pe1.run("ip link set dev PE1-eth0 master br101")
+    pe1.run("ip link set dev PE1-eth0 up")
+    # pe2
+    pe2.run("ip link add name br101 type bridge stp_state 0")
+    pe2.run("ip addr add 10.10.1.3/24 dev br101")
+    pe2.run("ip link set dev br101 addr aa:bb:cc:00:22:ff")
+    pe2.run("ip link set dev br101 master VRF-A")
+    pe2.run("ip link set dev br101 up")
+    pe2.run(
+        "ip link add vxlan101 type vxlan id 101 dstport 4789 local 10.30.30.30 nolearning"
+    )
+    pe2.run("ip link set dev vxlan101 master br101")
+    pe2.run("ip link set dev vxlan101 type bridge_slave neigh_suppress on learning off")
+    pe2.run("ip link set dev vxlan101 up")
+    pe2.run("ip link set dev PE2-eth1 master br101")
+    pe2.run("ip link set dev PE2-eth1 up")
+
+    ## Enable IPv4 Routing
+    p1.run("sysctl -w net.ipv4.ip_forward=1")
+    pe1.run("sysctl -w net.ipv4.ip_forward=1")
+    pe2.run("sysctl -w net.ipv4.ip_forward=1")
+
+    ## tell hosts to send GARP upon IPv4 addr assignment
+    host1.run("sysctl -w net.ipv4.conf.host1-eth0.arp_announce=1")
+    host2.run("sysctl -w net.ipv4.conf.host2-eth0.arp_announce=1")
+
+    ## Load FRR config on all nodes and start topo
+    router_list = tgen.routers()
+    for rname, router in router_list.items():
+        router.load_config(
+            TopoRouter.RD_ZEBRA, os.path.join(CWD, "{}/zebra.conf".format(rname))
+        )
+        router.load_config(
+            TopoRouter.RD_OSPF, os.path.join(CWD, "{}/ospfd.conf".format(rname))
+        )
+        router.load_config(
+            TopoRouter.RD_BGP, os.path.join(CWD, "{}/bgpd.conf".format(rname))
+        )
+    tgen.start_router()
+
+
+def teardown_module(mod):
+    "Teardown the pytest environment"
+    tgen = get_topogen()
+    tgen.stop_topology()
+
+
+def show_vni_json_elide_ifindex(pe, vni, expected):
+    output_json = pe.vtysh_cmd("show evpn vni {} json".format(vni), isjson=True)
+    if "ifindex" in output_json:
+        output_json.pop("ifindex")
+
+    return topotest.json_cmp(output_json, expected)
+
+
+def check_vni_macs_present(tgen, router, vni, maclist):
+    result = router.vtysh_cmd("show evpn mac vni {} json".format(vni), isjson=True)
+    for rname, ifname in maclist:
+        m = tgen.net.macs[(rname, ifname)]
+        if m not in result["macs"]:
+            return "MAC ({}) for interface {} on {} missing on {} from {}".format(
+                m, ifname, rname, router.name, json.dumps(result, indent=4)
+            )
+    return None
+
+
+def test_pe1_converge_evpn():
```

Comment:
should there be a wait here, or are we certain the table will have time to converge before we pull it?